bundle/Command/TagContentByTypesCommand.php

```diff
@@ -0,0 +1,198 @@
+                    function () use ($searchHit, $fieldIdentifiers): ?int {
+                        /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Content $content */
+                        $content = $searchHit->valueObject;
+
+                        $contentDraft = $this->repository->getContentService()->createContentDraft($content->contentInfo);
+                        $contentUpdateStruct = $this->repository->getContentService()->newContentUpdateStruct();
+
+                        foreach ($fieldIdentifiers as $fieldIdentifier) {
+                            if (!$this->hasField($content, $fieldIdentifier)) {
+                                continue;
+                            }
+
+                            if (!$content->getField($fieldIdentifier)->value instanceof TagFieldValue) {
+                                $this->style->error(sprintf('Field with identifier %s must be a type of eztags', $fieldIdentifier));
+                                return Command::FAILURE;
+                            }
+
+                            $alreadyAssignedTags = $content->getFieldValue($fieldIdentifier)->tags;
+                            $tagsToAssign = array_filter($alreadyAssignedTags, fn ($alreadyAssignedTag) => $this->getTag()->id !== $alreadyAssignedTag->id);
+                            $tagsToAssign[] = $this->getTag();
+                            $contentUpdateStruct->setField($fieldIdentifier, new TagFieldValue($tagsToAssign));
+
+                            break;
+                        }
+
+                        $this->repository->getContentService()->updateContent($contentDraft->versionInfo, $contentUpdateStruct);
+                        $this->repository->getContentService()->publishVersion($contentDraft->versionInfo);
+
+                        $this->style->progressAdvance();
+
+                        return null;
+                    }
+                );
+
+                if ($result === Command::FAILURE) {
+                    return Command::FAILURE;
+                }
+            }
+        }
+
+        $this->repository->commit();
+
+        $this->style->progressFinish();
+        $this->style->success('Tags assigned successfully');
+
+        return Command::SUCCESS;
+    }
+
+    protected function getParentLocation(): int
+    {
+        $parentLocation = $this->input->getOption('parent-location');
+
+        if (!is_numeric($parentLocation) || (int)$parentLocation < 1) {
+            throw new InvalidOptionException(
+                sprintf("Argument --parent-location must be an integer > 0, you provided '%s'", $parentLocation),
+            );
+        }
+
+        return (int)$parentLocation;
+    }
+
+    protected function getContentTypes(): array
+    {
+        return $this->parseCommaDelimited($this->input->getOption('content-types'));
+    }
+
+    protected function getTag(): Tag
+    {
+        return $this->tagsService->loadTag($this->getTagId());
```

Comment:
Not sure it makes sense to have this only as one line, and a separate method `getTagId` that is not being used anywhere else?

Reply:
It doesn't make sense, agreed.

Reply:
fixed in bd3915a6